### PR TITLE
Jvenstad/really hide system in role api

### PR DESCRIPTION
**Pull request by jonmv**
@tokle or @mpolden please review.

### CODE REVIEW

File: controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/role/Enforcer.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import com.yahoo.config.provision.SystemName;
+
+import java.net.URI;
+
+/**
+ * Checks whether {@link Role}s have the required {@link Privilege}s to perform {@link Action}s on given {@link java.net.URI}s.
+ *
+ * @author jonmv
+ */
+public class Enforcer {
```
Comment (mpolden) [inline controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/role/Enforcer.java]: enforcer

File: controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/role/Action.java [hunk 2]
```diff
@@ -18,11 +18,11 @@
     update,
     delete;
 
-    public static EnumSet<Action> all() {
+    static EnumSet<Action> all() {
```
Comment (mpolden) [inline controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/role/Action.java]: Why all the reduction in visibility?

File: controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/role/Enforcer.java (hunk 1 above)
Comment (jonmv) [reply controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/role/Enforcer.java]: Hehe. Not 100% sure about the name.

File: controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/role/Action.java (hunk 2 above)
Comment (jonmv) [reply controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/role/Action.java]: Hiding the parts that aren't meant to be used outside the package.

Comment (jonmv) [reply controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/role/Action.java]: The intended API is `Enforcer.allows`, and things to make `Role`s and  `Action`s with.

**Review by mpolden — APPROVED** [review]

**Pull request merged**